app/services/proofing/aamva/authentication_client.rb

```diff
@@ -2,28 +2,20 @@ module Proofing
   module Aamva
     class AuthenticationClient
       AAMVA_TOKEN_FRESHNESS_SECONDS = 28 * 60
+      AUTH_TOKEN_CACHE_KEY = 'aamva_api_auth_token'
 
-      class << self
-        attr_accessor :auth_token
-        attr_accessor :auth_token_expiration
-      end
-
-      def self.token_mutex
-        @token_mutex ||= Mutex.new
-      end
-
-      def fetch_token(config)
-        AuthenticationClient.token_mutex.synchronize do
-          if AuthenticationClient.auth_token.nil? || auth_token_expired?
-            send_auth_token_request(config)
-          end
-          AuthenticationClient.auth_token
+      def self.auth_token(config)
+        Rails.cache.fetch(
+          AUTH_TOKEN_CACHE_KEY,
+          skip_nil: true,
+          expires_in: AAMVA_TOKEN_FRESHNESS_SECONDS,
```

Comment:
oh i really like just having the cache handle the expiration

Reply:
There's also a chance this reduces the number of requests to AAMVA for tokens since the mutex would have been a local cache for an individual web server process, but moving it to Rails.cache (backed by Redis) will allow all instances to share the token.